### PR TITLE
[NoJira] update check-owners

### PR DESCRIPTION
**Pull request by matteo-hertel**
I got a bit carried away but this fixes the issue with `.` files we have during release, and I've added some async/await cause why not? 😄

### CODE REVIEW

File: scripts/npm/check-owners.js [hunk 1]
```diff
@@ -19,23 +19,80 @@
 /* @flow */
 
 /* eslint-disable no-console */
+/* eslint-disable no-use-before-define */
```
Comment (matteo-hertel) [inline scripts/npm/check-owners.js]: I do want hoisting in my JS!

Comment (matthewdavidson) [reply scripts/npm/check-owners.js]: Unfortunately the rule is there for a reason Matteo :)

Comment (matteo-hertel) [reply scripts/npm/check-owners.js]: In this particular instance I want to allow hoisting, it dramatically improve readability having the uniform function on top, opening the file will give a full overview of the business logic, implementation details are further down

Comment (georgegillams) [reply scripts/npm/check-owners.js]: I'm with @matthewdavidson  on this one. Maybe C# / Java devs would find this more readable.
If the business logic is really much lower-level than the overall business logic, it should probably be split into multiple files.

Comment (matteo-hertel) [reply scripts/npm/check-owners.js]: I completely disagree with that, that's not C# / Java at all, splitting the logic in multiple file doesn't really make sense here as won't be re-used by other scripts, at least not for now

**Comment by matteo-hertel** [comment]
I've decided to revert most of the changes I've done and just add the fix for the core issue of having the script picking up the dotfiles

**Pull request merged**